lib/_stream_readable.js

```diff
@@ -190,15 +190,17 @@ Readable.prototype.setEncoding = function(enc) {
   return this;
 };
 
-// Don't raise the hwm > 128MB
-const MAX_HWM = 0x800000;
-function roundUpToNextPowerOf2(n) {
-  if (n >= MAX_HWM) {
-    n = MAX_HWM;
+// Don't raise the hwm > 8MB
+const kHighWaterMarkBits = 23;
+const kHighWaterMark = 1 << kHighWaterMarkBits;
+
+function computeNewHighWaterMark(n) {
+  if (n >= kHighWaterMark) {
+    n = kHighWaterMark;
   } else {
     // Get the next highest power of 2
     n--;
-    for (var p = 1; p < 32; p <<= 1) n |= n >> p;
+    for (var p = 1; p < kHighWaterMarkBits; p <<= 1) n |= n >> p;
```

Comment:
FWIW, this could be simplified to `return 1 << 32 - Math.clz32(n);` but then it won't back-port to v2.x if we ever decide to cherry-pick it and do another release.  Probably not a major concern.